syntax = "proto3";

package order;

service OrderService {
  rpc CreateOrder (CreateOrderRequest) returns (OrderResponse);
  rpc GetOrder(GetOrderRequest) returns (OrderResponse);
  rpc UpdateOrder (UpdateOrderRequest) returns (OrderResponse);
}

message Order {
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 user_id = 4;
}

message CreateOrderRequest{
  string title = 1;
  string description = 2;
  int32 user_id = 3;
}

message GetOrderRequest{
  int32 id = 1;
}

message UpdateOrderRequest{
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 user_id = 4;
}

message OrderResponse {
  Order order = 1;
}
